src/components/view/ContactIntake/index.js

```diff
@@ -1,3 +1,497 @@
+            narcanWasTaken: false,
+            enrollment: '',
+            numberOfOthersHelping: 0,
+
+            // standard
+            uid: params.uid,
+            referral: null,
+            eventLocation: {
+                longitude: null,
+                latitude: null,
+            },
+        }
+
+        this.initialState = this.state;
+        this.submittedState = this.state;
+    }
+
+    // i'm sure we'll want to change names on the db in the future at some time
+    // or locally within state. so I'm abstracting this call to make it clear what data
+    // we send in event creation
+    packageFormDataForSubmission() {
+        const outreach = this.state.showOutreach ? {
+            syringesGiven: this.state.syringesGiven,
+            syringesTaken: this.state.syringesTaken,
+            narcanWasOffered: this.state.narcanWasOffered,
+            narcanWasTaken: this.state.narcanWasTaken,
+            enrollment: this.state.enrollment,
+            numberOfOthersHelping: this.state.numberOfOthersHelping,
+        } : null;
+
+         const standard = {
+             referral: this.state.referral,
+             location: this.state.eventLocation,
+             contactUid: this.state.uid,
+         }
+
+        const periodic = this.state.showPeriodic ? {
+            date: this.state.eventDate,
+            housingStatus: this.state.housingStatus,
+            hivStatus: this.state.hivStatus,
+            isInCareForHiv: this.state.isInCareForHiv,
+            hepCStatus: this.state.hepCStatus,
+            isInCareForHepC: this.state.isInCareForHepC,
+            healthInsurer: this.state.healthInsurer,
+            primaryDrug: this.state.primaryDrug,
+            didOdLastYear: this.state.didOdLastYear,
+            didSeeOdLastYear: this.state.didSeeOdLastYear,
+            hasHealthInsurance: this.state.hasHealthInsurance,
+            otherDrugs: this.state.otherDrugs,
+        } : null;
+
+        const contactData = this.state.showNewContactQuestions ? {
+            newContactDate: this.state.newContactDate,
+            contactDateOfBirth: this.state.contactDateOfBirth,
+            contactGenderIdentity: this.state.contactGenderIdentity,
+            contactEthnicity: this.state.contactEthnicity,
+            contactIsHispanic: this.state.contactIsHispanic,
+            contactCountryOfBirth: this.state.contactCountryOfBirth,
+            contactAgeOfFirstInjection: this.state.contactAgeOfFirstInjection,
+        } : null;
+
+        // dirty check to only submit data for visible forms
+        let prunedEventData = {
+           ...outreach,
+           ...standard,
+           ...periodic,
+           ...contactData,
+        };
+        return prunedEventData;
+    }
+
+    submitForm() {
+        // TODO: Ultimately should change these cases to prompts, not alert; but React errors for now
+        // TODO: what about validation?
+        if (this.initialState == this.state) {
```

Comment:
Form "cleanliness" should be able to be handled by redux-form. You could use the [pmc-auction](https://github.com/evandana/pmc-auction) site as reference on the CreateAuction or CreateRaffle forms.